pkg/client/userd/trafficmgr/traffic_manager.go

```diff
@@ -487,6 +487,10 @@ func (tm *TrafficManager) getInfosForWorkloads(
 }
 
 func (tm *TrafficManager) WatchWorkloads(c context.Context, wr *rpc.WatchWorkloadsRequest, stream WatchWorkloadsStream) error {
+	c, cancel := context.WithCancel(c)
+	// We need to make sure the subscription ends when we leave this method, since this is the one consuming the snapshotAvailable channel.
+	// Otherwise, the goroutine that writes to the channel will leak.
+	defer cancel()
 	snapshotAvailable := tm.wlWatcher.subscribe(c)
```

Comment:
This won't fly. We can't mix wlWatcher's using this context with wlWatcher's using the session context. This has to use the `sessionContext`. It will linger, for sure, and we might want to address that at some point.